sdk/python/feast/pyspark/abc.py

```diff
@@ -451,18 +482,47 @@ def get_arguments(self) -> List[str]:
             "online",
         ]
 
+    def get_job_hash(self) -> str:
+        job_json = json.dumps(
+            {"source": self._source, "feature_table": self._feature_table},
+            sort_keys=True,
+        )
+        return hashlib.md5(job_json.encode()).hexdigest()
+
 
 class BatchIngestionJob(SparkJob):
     """
     Container for the ingestion job result
     """
 
+    def to_proto(self) -> JobProto:
+        job = super().to_proto()
```

Comment:
Not really important, but I think start & end date also should be used in hash calculation for batch ingestion if we want to represent job identity by it, even though we wan't run this job in streaming loop